pkg/statedb/table.go

```diff
@@ -13,11 +13,14 @@ import (
 	"github.com/cilium/cilium/pkg/statedb/index"
 )
 
+// NewTable creates a new table with given name and indexes.
+// Panics if indexes are malformed which allows this to be used at
```

Comment:
Is this worth the risk of a runtime panic?

As far as I can tell, it's now possible to create and register tables at an arbitrary point at runtime, and there might even be a legitimate use-case for it in the future. If that's so, then I'd much rather have to deal with an error than have a potential panic, even if it leads to slightly uglier code in the tests.

Reply:
That is a good point, this does invite table creation after the constructor/start hooks where panics are less ideal. Even though its not how we intend it, that is going to be inevitable.